### DISCUSSION TITLE
3 column layout

### DISCUSSION THREAD

**Discussion by dimonf** [sxyazi/yazi#1146] [4. Ideas]
First, thank you for great app.
I am thinking, it would be great if I can change layout a bit if I need to. I am using tiling WM and sometime need to explore files in one of terminal window, which may accommodate 1/4 of the screen area. While **yazi** resource-wise is small enough not to think twice to fire it up as other tools from standard unix toolbox, it's standard 3 columns use of screen area in certain circumstances is far from optimum. If, say, left two columns can form a stack, placing section with lower dir content below the section with content of parent dir? 

![out-less](https://github.com/sxyazi/yazi/assets/842666/7c6030d4-57c4-4d22-a228-7824e7f9c580)

**Comment by lpnh** [comment]
edit: update code to Yazi 0.3

This is not exactly what you are asking for, but I created a custom plugin for myself for cases like this. It will hide the first column and resize the other two:

![yazi-custom-layout](https://github.com/sxyazi/yazi/assets/90577992/e9842a30-12de-4b6c-9340-aafaca585f47)

Here's how you can achieve it:

- Create a custom plugin, e.g. `custom-layout.yazi/init.lua`

```lua
local function entry(st)
	if st.old then
		Tab.layout, st.old = st.old, nil
	else
		st.old = Tab.layout
		Tab.layout = function(self)
			self._chunks = ui.Layout()
				:direction(ui.Layout.HORIZONTAL)
				:constraints({
					ui.Constraint.Percentage(0),
					ui.Constraint.Percentage(25),
					ui.Constraint.Percentage(75),
				})
				:split(self._area)
		end
	end
	ya.app_emit("resize", {})
end

return { entry = entry }
```

- Add it to your Yazi plugins directory, e.g. `~/.config/yazi/plugins/custom-layout.yazi`

- Bind it to a unused key, e.g. `F`, in your `keymap.toml`

```
[[manager.prepend_keymap]]
on = ["F"]
run = "plugin --sync custom-layout"
```

That's it. You can further modify the columns ratio by changing the `ui.Constraint.Percentage()` values:

```
ui.Constraint.Percentage(0),
ui.Constraint.Percentage(25),
ui.Constraint.Percentage(75),
```

The inspiration for this came from the `max-preview` plugin: https://github.com/yazi-rs/plugins/tree/main/max-preview.yazi

**Reply by dimonf** [reply]
Thank you,  pretty neat! I believe your solution is good enough to finish discussion :)

**Comment by sxyazi** [comment]
Maybe you could also try setting [`ratio = [0, 3, 2]`](https://yazi-rs.github.io/docs/configuration/yazi#manager.ratio) and then use this plugin to preview the directory tree: https://github.com/sharklasers996/eza-preview.yazi

![eza-preview.yazi](https://github.com/sharklasers996/eza-preview.yazi/blob/master/tree.png?raw=true)

**Reply by dimonf** [reply]
Thank you for the idea. I may be 100% wrong, but for me, yazi appeal is not in swiss knife endless versatility - we obviously have shell for this.  Yazi is made of well defined blocks, simple and functional. I'd prefer it to stay this way.  My point was to have limited alternative layouts which serve single only purpose: better utilization of screen asset.  As lpnh suggested before, simple removal of first column does the trick. The reason I've suggested to stack left 2 columns vertically is this: both blocks has similar type of rows, where lines for dirs and filenames can be long. Almost all tiling windows managers can quickly switch between vertical and horizontal placement of adjacent windows.
